### PR TITLE
Break up the rather large operations.go into multiple files

### PR DESCRIPTION
**Pull request by phlogistonjohn**
### What does this PR achieve? Why do we need it?

The file operations.go has grown to the point where it is too large. This PR splits up the file such that operations.go only retains the most generic functions & types. New files for volume operations, block volume operations, and device operations are added. Similarly, the operations_test.go is broken up in a similar manner. The mutli cluster error is put in a new file as well.

This should make working with the operations & their tests simpler in the future.

### Notes for the reviewer

No substantive changes were made in these changesets, save to get the imports correct for the new files. Every other line in the new files should be exactly as it was in operations.go.

This is preparatory work for #1401. This series is based on top of #1400 please review and merge that first.

### CODE REVIEW

File: apps/glusterfs/operations_block_volume.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+package glusterfs
+
+import (
+	"fmt"
+
+	"github.com/heketi/heketi/executors"
+	wdb "github.com/heketi/heketi/pkg/db"
+
+	"github.com/boltdb/bolt"
```
Comment (Madhu-1) [inline apps/glusterfs/operations_block_volume.go]: this import order should be changed?

importing third party lib should be the second set of import

File: apps/glusterfs/operations_block_volume.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+)
+
+// BlockVolumeCreateOperation  implements the operation functions used to
+// create a new volume.
+type BlockVolumeCreateOperation struct {
+	OperationManager
+	noRetriesOperation
+	bvol *BlockVolumeEntry
+	//vol *VolumeEntry
```
Comment (Madhu-1) [inline apps/glusterfs/operations_block_volume.go]: commented code

File: apps/glusterfs/operations_device.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"fmt"
+
+	"github.com/heketi/heketi/executors"
```
Comment (Madhu-1) [inline apps/glusterfs/operations_device.go]: which import format we need to follow
```
import (
	"errors"
	"fmt"
	"os"
	"strconv"

	"github.com/lpabon/godbc"

	"github.com/heketi/heketi/executors/cmdexec"
)
```
and
```
import (
	"bytes"
	"encoding/gob"
	"encoding/json"
	"net/http"
	"time"

	"github.com/heketi/heketi/pkg/glusterfs/api"

	"github.com/boltdb/bolt"
	"github.com/lpabon/godbc"
)
```

File: apps/glusterfs/operations_block_volume.go (hunk 1 above)
Comment (phlogistonjohn) [reply apps/glusterfs/operations_block_volume.go]: Sure, but lets do that in a follow up commit. I've been trying to use that approach personally but right now heketi as a project does not enforce that as a general rule (yet).

File: apps/glusterfs/operations_block_volume.go (hunk 2 above)
Comment (phlogistonjohn) [reply apps/glusterfs/operations_block_volume.go]: This commit was a pure move and the comment existed previously. Will clean it up at a later time.

**Comment by obnoxxx** [comment]
@phlogistonjohn I did the rebase on #1400 for you ;-)

**Pull request merged**